bindings/serialize.cc

```diff
@@ -0,0 +1,397 @@
+  }
+  virtual int64_t getFileID() { return node->script_id; }
+  virtual int64_t getLineNumber() { return node->line_number; }
+  virtual int64_t getColumnNumber() { return node->column_number; }
+  virtual std::vector<Sample> getSamples(Profile &p,
+                                         std::deque<uint64_t> stack) {
+    std::vector<Sample> samples;
+    for (std::vector<AllocationProfile::Allocation>::iterator allocation =
+             node->allocations.begin();
+         allocation != node->allocations.end(); ++allocation) {
+      std::vector<Label> labels = {Label(p.getStringId("allocation"), 0,
+                                         allocation->size,
+                                         p.getStringId("bytes"))};
+      std::vector<int64_t> vals = {
+          allocation->count,
+          static_cast<int64_t>(allocation->size * allocation->count)};
+      Sample s = Sample({stack.begin(), stack.end()}, vals, labels);
+      samples.push_back(s);
+    }
+    return samples;
+  }
+};
+
+class TimeNode : public Node {
+ private:
+  const CpuProfileNode *node;
+  int samplingIntervalMicros;
+
+ public:
+  TimeNode(const CpuProfileNode *node, int samplingIntervalMicros) {
+    this->node = node;
+    this->samplingIntervalMicros = samplingIntervalMicros;
+  }
+  virtual std::string getName() {
+    return *String::Utf8Value(node->GetFunctionName());
+  }
+  virtual std::string getFilename() {
+    return *String::Utf8Value(node->GetScriptResourceName());
+  }
+  virtual int64_t getFileID() { return node->GetScriptId(); }
+  virtual int64_t getLineNumber() { return node->GetLineNumber(); }
+  virtual int64_t getColumnNumber() { return node->GetColumnNumber(); }
+  virtual std::vector<Sample> getSamples(Profile &p,
+                                         std::deque<uint64_t> stack) {
+    std::vector<Sample> samples;
+    int64_t hitCount = node->GetHitCount();
+    std::vector<int64_t> vals = {hitCount, hitCount * samplingIntervalMicros};
+    Sample s = Sample({stack.begin(), stack.end()}, vals, std::vector<Label>());
+    samples.push_back(s);
+    return samples;
+  }
+};
+
+struct TimeEntry {
+  const CpuProfileNode *node;
+  std::deque<uint64_t> stack;
+};
+
+std::vector<char> serializeTimeProfile(CpuProfile *profile,
+                                       int64_t samplingIntervalMicros,
+                                       int64_t startTimeNanos) {
+  int64_t durationNanos =
+      (profile->GetEndTime() - profile->GetStartTime()) * 1000;
+
+  Profile p = Profile("wall", "microseconds", samplingIntervalMicros,
+                      startTimeNanos, durationNanos);
+  p.addSampleType("sample", "count");
+  p.addSampleType("wall", "microseconds");
+
+  // Add root to entries
+  const CpuProfileNode *root = profile->GetTopDownRoot();
+  std::vector<TimeEntry> entries;
+  int32_t count = root->GetChildrenCount();
+  for (int32_t i = 0; i < count; i++) {
+    TimeEntry child;
+    child.node = root->GetChild(i);
```

Comment:
I think this could be `TimeEntry child = { root->GetChild(i) };`

Reply:
Done.